Introduce Admin API chapter (#1860)
PrestaShop/docs: admin-api/context.md

---
title: Context of the Admin API
menuTitle: Context
weight: 1
---

# Context of the Admin API

## Why a new Admin API?

The new admin API is here to replace and improve the existing [Webservice API]({{< relref "../webservice" >}}) that has a few drawbacks:

> - Permission and security management limited

Reviewer: ```suggestion
- Limited permission and security management
```

- Performs CRUD operations without much intelligence
- A bit old compared to the new web standards
- Legacy code, not tested, hard to maintain, scary to modify
- Based on ObjectModel which is supposed to disappear one day

### The goals of the new Admin API

The modern API is meant to answer these new needs:

> - Bring a more secure API

Reviewer 2: ```suggestion
- to bring a more secure API
```

> - Allows more tracking in actions (know who/what is doing what)

Reviewer 2: ```suggestion
- to allow more tracking in actions (know who/what is doing what)
```

Reviewer 2: and the lines below

- Give more intelligence to the API than basic CRUD
- Be more in tune with the modern standards
- Extendable by modules

> To reach these goals we decided an implementation based on:

Reviewer 2: ```suggestion
To reach these goals, we decided an implementation based on:
```
